doc/platform_api/CMIS_Diagnostic_Monitoring_Overview_in_SONiC.md

```diff
@@ -0,0 +1,2375 @@
+- `rxoutput_status_hostlane{lane_num}`
+- `config_state_hostlane{lane_num}`
+- `dpdeinit_hostlane{lane_num}`
+
+The transceiver status flags, change count, and their set/clear time for the following fields are updated in the `redis-db` during a link change event:
+
+- `datapath_firmware_fault`
+- `module_firmware_fault`
+- `module_state_changed`
+- `txfault{lane_num}`
+- `rxlos{lane_num}`
+- `txlos_hostlane{lane_num}`
+- `txcdrlol_hostlane{lane_num}`
+- `tx_eq_fault{lane_num}`
+- `rxcdrlol{lane_num}`
+
+**Example:**
+
+The following fields related to `datapath_firmware_fault` are updated in the `redis-db` during a link change event:
+
+- `datapath_firmware_fault` in `TRANSCEIVER_STATUS` table
+- `datapath_firmware_fault` in `TRANSCEIVER_STATUS_FLAG_CHANGE_COUNT` table
+- `datapath_firmware_fault` in `TRANSCEIVER_STATUS_FLAG_SET_TIME` table
+- `datapath_firmware_fault` in `TRANSCEIVER_STATUS_FLAG_CLEAR_TIME` table
+
+#### 4.2.3 Details of Flag Analysis of Tables
+
+**Note**: For simplicity, this section uses DOM as an example. However, the same analysis is applicable for VDM and Status related flags as well.
+
+**Purpose of Flag Analysis:**
+
+The purpose of flag analysis is to track the status of various parameters and to count the number of times each DOM flag has changed. It also records the timestamp when each DOM flag was set and cleared.
+
+**Tables Used for Flag Analysis:**
+
+- `TRANSCEIVER_DOM_FLAG`: This table stores flags indicating the status of various DOM parameters.
+- `TRANSCEIVER_DOM_FLAG_CHANGE_COUNT`: This table keeps a count of how many times each DOM flag has changed. Upon initialization, the count is set to 0.
+- `TRANSCEIVER_DOM_FLAG_SET_TIME`: This table records the timestamp (in local timezone) when each DOM flag was set. The timestamp is recorded in the format `Day Mon DD HH:MM:SS YYYY`. During initialization, the timestamp is set to `never` if the flag is not set. Since SONiC does not support flag-based interrupt handling, the timestamp refers to either:
+  - The timestamp at which the link status was changed, or
+  - The polling event timestamp if the flag was set during the routine polling by the `DomInfoUpdateTask` thread.
+- `TRANSCEIVER_DOM_FLAG_CLEAR_TIME`: This table records the timestamp (in local timezone) when each DOM flag was cleared. The timestamp is recorded in the format `Day Mon DD HH:MM:SS YYYY`. During initialization, the timestamp is set to `never` if the flag is set. Since SONiC does not support flag-based interrupt handling, the timestamp refers to either:
+  - The timestamp at which the link status was changed, or
+  - The polling event timestamp if the flag was cleared during the routine polling by the `DomInfoUpdateTask` thread.
+
+**Example of Table Updates:**
+
+- **TRANSCEIVER_DOM_FLAG_CHANGE_COUNT:**
+  - Each time a flag in the `TRANSCEIVER_DOM_FLAG` table changes (either set or cleared), the corresponding count in this table is incremented.
+- **TRANSCEIVER_DOM_FLAG_SET_TIME:**
```

Comment:
@mihirpat1 

1. this is the `FIRST` time dom finds the flag is set either as part of link down event or normal periodic?
2. What if the flag remains set, do we update the time next polling?

Reply:
@mihirpat1 can you modify the wording that flag set time recorded corresonds to the time when Xcvrd polls and its not the actual time when module sets.

Reply:
@prgeor I have addressed this.